### PR TITLE
Partial select checkbox

### PR DESCRIPTION
**Pull request by ChaoLiangSuper**
## Description
Fixes: https://github.com/Adslot/adslot-ui/issues/796
- Rewrite checkbox component
- Deprecate `iCheck` classes
- Added new svgs for checkbox icons
- Added partial state
- Change onChange API arguments (Because we used to get value, but never used the value)
   Change from `onChange(event, name)` to `onChange(nextState, name, value)`
- State cycle for the checkbox will be 
   true -> false,
   false -> true,
   'partial' -> false,
- Add `size` prop to set the size of the checkbox (default: 16)
- ~Reduce DOM elements for only using checkbox without label~ 
(Revert this change, cause if there is not `label` the input will not catch the `onChange` event)

- [x] Test and coverage
- [x] Update checkbox group 

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Screenshots (if appropriate):
screen shot 2019-02-06 at 12 35 12 pm

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: config/dev-cold.js [hunk 1]
```diff
@@ -18,17 +18,6 @@ module.exports = merge(commonConfig, {
     filename: 'app.js',
     publicPath,
   },
-  module: {
```
Comment (ChaoLiangSuper) [inline config/dev-cold.js]: Using `url-loader` to handle images

File: src/components/adslot-ui/Checkbox/styles.scss [hunk 2]
```diff
@@ -1,3 +1,84 @@
...
+  }
+
+  .checkbox-component-icon {
+    display: inline-block;
+    background-repeat: no-repeat;
+    background-size: contain;
+    background-image: url('~styles/icons/checkbox/empty.svg');
+  }
+
+  &:hover:not(.disabled):not(.checked):not(.partial-checked) .checkbox-component-icon {
```
Comment (nimishjha) [inline src/components/adslot-ui/Checkbox/styles.scss]: If we're using SVGs, can the disabled states have a little more contrast? Right now I can barely see the difference between `checked disabled` and `partial disabled`.

Comment (ChaoLiangSuper) [reply src/components/adslot-ui/Checkbox/styles.scss]: Those SVGs are from UX, so need to check with UX team.

**Comment by ChaoLiangSuper** [comment]
@lightbringer1991 I don't think so, we have several PR which already marked `breaking` and I also checked with other people they are not block by this. So will do a 27 release right after this. So no need to have a feature branch.

**Comment by ChaoLiangSuper** [comment]
Design reviewed, screenshot updated.

**Pull request merged**